tests/testthat/test-pstr_add_reductions.R

```diff
@@ -6,3 +6,75 @@ test_that("hasn't changed", {
   out <- format_robust_snapshot(with_reductions)
   expect_snapshot(out)
 })
+
+test_that("throws an error when the user do not give enough parameters", {
+  expect_error(pstr_add_reductions())
+  expect_error(pstr_add_reductions(pstr_toy_companies()))
+  expect_error(pstr_add_reductions(pstr_toy_companies(), pstr_toy_ep_weo()))
+})
+
+test_that("throws an error when the user do not give the parameters in the right order", {
+  expect_error(pstr_add_reductions(pstr_toy_ep_weo(), pstr_toy_companies(), pstr_toy_weo_2022()))
+  expect_error(pstr_add_reductions(pstr_toy_weo_2022(), pstr_toy_companies(), pstr_toy_ep_weo()))
+})
+
+test_that("returns a tibble data frame", {
+  expect_s3_class(
+    pstr_add_reductions(pstr_toy_companies(), pstr_toy_ep_weo(), pstr_toy_weo_2022()),
+    "tbl_df"
+  )
+})
+
+test_that("adds twelve new columns: `EP_categories_id`, `EP_group`, `weo_product_mapper`,
+          `weo_flow_mapper`, `publication`, `scenario`, `region`, `category`,
+          `unit`,`year`, `value`, `reductions`", {
+  # TODO : Can we make this less redundant ?
+  expect_false(hasName(pstr_toy_companies(), "EP_categories_id"))
+  expect_false(hasName(pstr_toy_companies(), "EP_categories_id"))
+  expect_false(hasName(pstr_toy_companies(), "EP_group"))
+  expect_false(hasName(pstr_toy_companies(), "weo_product_mapper"))
+  expect_false(hasName(pstr_toy_companies(), "weo_flow_mapper"))
+  expect_false(hasName(pstr_toy_companies(), "publication"))
+  expect_false(hasName(pstr_toy_companies(), "scenario"))
+  expect_false(hasName(pstr_toy_companies(), "region"))
+  expect_false(hasName(pstr_toy_companies(), "category"))
+  expect_false(hasName(pstr_toy_companies(), "unit"))
+  expect_false(hasName(pstr_toy_companies(), "year"))
+  expect_false(hasName(pstr_toy_companies(), "value"))
+  expect_false(hasName(pstr_toy_companies(), "reductions"))
```

Comment:
This doesn't seem to belong here but to belong in tests/testthat/test-pstr_utils.R. You can test the names the toy data must have and then you know it has nothing else than that.